trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/TrinoLogsResource.java

```diff
@@ -0,0 +1,242 @@
+                requireNonNull(eventId, "eventId is null");
+                requireNonNull(message, "message is null");
+            }
+        }
+    }
+
+    @Inject
+    public TrinoLogsResource(RequestLoggerController loggerController, ObjectMapper objectMapper)
+    {
+        this.loggerController = requireNonNull(loggerController, "loggerController is null");
+        this.objectMapper = requireNonNull(objectMapper, "objectMapper is null");
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response processRequest(@Context Request request, @HeaderParam("X-Amz-Target") String target)
+    {
+        String command = Splitter.on('.').trimResults().limit(2).splitToList(target).getLast().toLowerCase(Locale.ROOT);
+        return switch (command) {
+            // filter/get log events are basically the same.
+            // We've made GetLogEventsRequest a union of both and one method can satisfy both
+            case "filterlogevents", "getlogevents" -> getLogEvents(request);
+            default -> Response.status(BAD_REQUEST).build();
+        };
+    }
+
+    private Response getLogEvents(Request request)
+    {
+        byte[] bytes = request.requestContent().standardBytes().orElseThrow(() -> new WebApplicationException(BAD_REQUEST));
+        GetLogEventsRequest getLogEvents;
+        try {
+            getLogEvents = objectMapper.readValue(bytes, GetLogEventsRequest.class);
+        }
+        catch (IOException e) {
+            throw new WebApplicationException(e, BAD_REQUEST);
+        }
+
+        // TODO - support filter patterns
+
+        boolean startFromHead = getLogEvents.startFromHead.orElse(false);
+        Predicate<SaveEntry> predicate = entry -> startTimePasses(getLogEvents, entry) &&
+                endTimePasses(getLogEvents, entry) &&
+                logStreamNamesPasses(entry, getLogEvents);
+        int limit = getLogEvents.limit.orElse(Integer.MAX_VALUE);
+
+        List<SaveEntry> filteredEntries = loggerController.savedEntries(startFromHead, predicate);
+
+        record StartIndex(TokenType tokenType, int index) {}
+
+        // next token is merely an index
+        StartIndex startIndex = getLogEvents.nextToken().map(nextToken -> {
+            try {
+                TokenType tokenType;
+                if (nextToken.startsWith(TokenType.FORWARDS.prefix)) {
+                    tokenType = TokenType.FORWARDS;
+                }
+                else if (nextToken.startsWith(TokenType.BACKWARDS.prefix)) {
+                    tokenType = TokenType.BACKWARDS;
+                }
+                else {
+                    throw new WebApplicationException("Invalid nextToken", BAD_REQUEST);
+                }
+                return new StartIndex(tokenType, Integer.parseInt(nextToken.substring(2)));
+            }
+            catch (NumberFormatException _) {
+                throw new WebApplicationException("Invalid nextToken", BAD_REQUEST);
```

Comment:
In `processRequest` we return `Response.status(BAD_REQUEST).build()` instead of throwing exceptions.

Should we do the same thing here?

Reply:
That would be cumbersome. Throwing `WebApplicationException` is exactly the same as returning a Response.